src/DiscreteSystem.ts

```diff
@@ -137,4 +138,18 @@ export class DiscreteSystem<TShared, TInternal> implements System {
         }
         return state;
     }
+
+    /**
+     * Sets new shared (parameter) values for all groups in the system.
+     * @param newShared Updated parameter values. The length must match the existing number of groups in the system.
+     */
+    public updateShared(newShared: TShared[]) {
+        if (newShared.length !== this._shared.length) {
+            throw new DustParameterError(
+                "New shared value must be same length as previous value. " +
+                    `Expected ${this._shared.length} but got ${newShared.length}.`
+            );
+        }
+        this._shared = newShared;
```

Comment:
This needs to do a bit more than this, because not all parameters might be updated - it's not a full replacement.